Updated documentation to Doxygen v1.9.2 and new layout.

vovamarch opened the pull request:
Updated Doxygen documentation build to use v1.9.2
Updated layout.
Minor content improvements

vovamarch committed:
Updated documentation to Doxygen v1.9.2 and new layout.

RobertRostohar reviewed (changes requested):
In general migration to newer doxygen looks ok.

The `gen_doc.sh` has an issue when called initially (after checkout).
It shows the following error:
`Output directory '../../Documentation/General' does not exist and cannot be created`

`header.html.bak` should be removed.

Now also the pack generator script needs to be updated to use the new `gen_doc.sh`. After that the `genDoc.bat` and `doxygen_general.bat` can be removed,

RobertRostohar commented on DoxyGen/General/src/introduction.txt:
```diff
@@ -1,43 +1,34 @@
+\b I2C/                   | \ref driver_I2C driver implementations
+\b NAND/                  | \ref driver_NAND driver implementations
+\b SPI/                   | \ref driver_SPI driver implementations
+\b Utilities/             | Utilities for generating the pack
```
`Utilities` should be removed

vovamarch replied:
Removed with the latest commit.

vovamarch committed:
Addressed review comments.

vovamarch commented:
Thanks @RobertRostohar!
- updated gen_doc.sh to create the Documentation/ folder if it is  absent.
- removed header.html.bak

I suggest that the pack generation with bash is done separately, and having the documentation working helps with that. Doc publishing can be then also automated with GitHub actions.
